src/minirest_handler.erl

```diff
@@ -16,29 +16,36 @@
 
 -export([init/2]).
 
--export([reply/2]).
-
 -include("minirest_http.hrl").
 
 -include("minirest.hrl").
 
 %%==============================================================================================
 %% cowboy callback init
 init(Request0, State)->
-    Response = handle(Request0, State),
-    Request = reply(Response, Request0),
+    {Response, Handler} = handle(Request0, State),
+    Request = reply(Response, Request0, Handler),
     {ok, Request, State}.
 
 %%%==============================================================================================
 %% internal
 handle(Request, State) ->
     Method = cowboy_req:method(Request),
-    case maps:get(Method, State, undefined) of
-        undefined ->
-            {?RESPONSE_CODE_METHOD_NOT_ALLOWED, #{<<"allow">> => [maps:keys(State)]}, <<"">>};
-        Handler ->
-            do_auth(Request, Handler)
-    end.
+    Handler = maps:get(Method, State, maps:keys(State)),
```

Comment:
Nit： Maybe use maps:find instead of maps:get.
I prefer that the function return the same type.

Don't over-addict the pattern-match...

Reply:
changed